### PR TITLE
Add support to block traffic to an application

### PR DESCRIPTION
**Pull request by chaitanyaenr**
### Description
This commit enables users to simulate a downtime of an application
by blocking the traffic for the specified duration to see how
it/other components communicating with it behave in case of downtime.

### CODE REVIEW

File: docs/application_outages.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+### Application outages
+Scenario to block the traffic ( Ingress/Egress ) of an application matching the labels for the specified duration of time to understand the behavior of the service/other services which depend it during the downtime. This helps with the planning the requirements accordingly be it improving the timeouts or tweaking the alerts etc.
+
+##### Sample scenario config
+```
```
Comment (paigerube14) [inline docs/application_outages.md:5]: Would be nice to have a sample here with filled in parameters. The pod_selector variable is a little confusing and in a different format than normal so unsure what to put.

**Comment by paigerube14** [comment]
On first run I get the error:
```
the path "kraken-network-policy.yaml" does not exist
2021-09-29 20:30:43,909 [ERROR] Failed to run kubectl create -f kraken-network-policy.yaml -n openshift-authentication, error: Command 'kubectl create -f kraken-network-policy.yaml -n openshift-authentication' returned non-zero exit status 1.

```

File: docs/application_outages.md (hunk 1 above)
Comment (chaitanyaenr) [reply docs/application_outages.md:5]: Right, updated the docs and sample configuration. Thanks.

**Comment by chaitanyaenr** [comment]
> On first run I get the error:
> 
> ```
> the path "kraken-network-policy.yaml" does not exist
> 2021-09-29 20:30:43,909 [ERROR] Failed to run kubectl create -f kraken-network-policy.yaml -n openshift-authentication, error: Command 'kubectl create -f kraken-network-policy.yaml -n openshift-authentication' returned non-zero exit status 1.
> ```

Should be fixed now. PTAL. Thanks.

File: kraken/application_outage/actions.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# block the traffic for the specified duration
+def run(scenarios_list, config, wait_duration):
+    failed_post_scenarios = ""
+    for app_outage_config in scenarios_list:
+        if len(app_outage_config) > 1:
+            with open(app_outage_config, "r") as f:
+                app_outage_config_yaml = yaml.full_load(f)
+                scenario_config = app_outage_config_yaml["application_outage"]
+                namespace = scenario_config["namespace"]
+                pod_selector = scenario_config["pod_selector"]
```
Comment (paigerube14) [inline kraken/application_outage/actions.py:19]: Can we add in a default value here in case the user doesn't want to set a pod selector like '{}'

**Review by paigerube14 — COMMENTED** [review]
LGTM other than default for pod_selector

File: kraken/application_outage/actions.py (hunk 2 above)
Comment (chaitanyaenr) [reply kraken/application_outage/actions.py:19]: Good idea, updated it. Thanks.

Comment (paigerube14) [reply kraken/application_outage/actions.py:19]: Are we able to do it like the parameters in the main run_kraken file? This way if the pod_selector key isn't found in the file 
ex.) chaos_scenarios = config["kraken"].get("chaos_scenarios", [])

I'm getting the below when commenting out the pod_selector line in the scenario file

```
    pod_selector = scenario_config["pod_selector"]
KeyError: 'pod_selector'
```

**Review by paigerube14 — APPROVED** [review]
LGTM

**Pull request merged**